### ISSUE TITLE
開発開始

### ISSUE THREAD

**Issue by Kalorie560** [Kalorie560/OptimalFert#2]
@claude

以下の要件の開発をしてください。

あなたは、Kaggleコンペティションの課題解決を支援するAIアシスタントです。これから、以下のKaggleコンペティションに対する包括的な開発プロジェクトを実行してください。

**コンペティション名:** Playground Series S5E6
(元URL: `https://www.kaggle.com/competitions/playground-series-s5e6/`)

**コンペティション詳細情報 (一般的なPlayground Seriesの傾向と仮定に基づく):**

* **目的:** 提供される複数の特徴量から、ターゲット変数を予測する。多くの場合、これは分類問題（例：二値分類）または回帰問題です。ここでは**二値分類問題**を仮定して進めます。
* **データファイル:**
    * `train.csv`: 学習用のデータセット。ID、複数の特徴量カラム、およびターゲット変数カラムが含まれます。
    * `test.csv`: 評価用のデータセット。ID、複数の特徴量カラムが含まれますが、ターゲット変数は含まれません。
    * `sample_submission.csv`: 提出ファイルの正しいフォーマットを示すサンプルファイル。通常、IDカラムと予測されたターゲット変数カラムが含まれます。
* **特徴量:**
    * `id`: 各行の一意な識別子。
    * その他多数の特徴量: これらは通常、数値（連続値または離散値）およびカテゴリカルな特徴量が混在しています。Playground Seriesでは匿名化されていることが多いです（例: `feature_1`, `feature_2`, ..., `feature_N`）。
    * **（AIへの指示）データ読み込み後、各特徴量のデータ型（数値、カテゴリなど）を特定し、適切に処理してください。**
* **ターゲット変数 (train.csv内):**
    * 名前は `target` (または類似の名前) であると仮定します。
    * 二値分類の場合、通常 0 または 1 の値を取ります。
* **評価指標:**
    * Playground Seriesの分類問題では、**ROC AUC (Area Under the Receiver Operating Characteristic Curve)** が一般的に使用されます。この指標で評価されるものと仮定します。
    * **（AIへの指示）モデルの評価およびハイパーパラメータチューニングの際には、このROC AUCを最大化することを目指してください。**
* **提出ファイルのフォーマット (`submission.csv`):**
    * `sample_submission.csv` に厳密に従う必要があります。
    * 通常、2つのカラム: `id` と `target` (または指定されたターゲット名)。
    * `id` カラムには `test.csv` の `id` をそのまま使用します。
    * `target` カラムには、各 `id` に対する予測確率（ROC AUCの場合）または予測クラスを記述します。二値分類でROC AUCが評価指標の場合、**陽性クラスに属する確率 (0.0から1.0の間の浮動小数点数)** を提出することが一般的です。
    * **（AIへの指示）提出ファイルのフォーマット（カラム名、IDの順序、値の範囲など）を`sample_submission.csv`とコンペの指示で再確認し、厳密に遵守してください。**

**プロジェクト全体の目標:**

1.  上記のコンペティション情報を基に、高精度な予測モデル（二値分類、ROC AUC最適化）を構築する。
2.  ClearMLを使用して学習プロセス全体を追跡・管理する。
3.  開発したモデルを使用して、特徴量を入力すると予測結果（陽性クラスの確率）を出力するウェブアプリケーションを作成する。
4.  日本語と英語で、詳細なREADMEファイルを作成する。
5.  コンペティションの規定に厳密に従ったSubmissionファイルを生成する。

**開発ステップと要件詳細:**

**1. プロジェクト設定と環境準備:**

* **ClearML連携:**
    * ClearML SDKを導入し、すべての実験（データ前処理、モデル学習、評価など）の履歴（ハイパーパラメータ、メトリクス（特にROC AUC）、生成されたモデルファイル、デバッグサンプルなど）をClearMLサーバーに記録・監視できるように設定してください。
    * ClearMLのAPIキーは、プロジェクトルートに配置する`config.yaml`ファイルから読み込むように実装してください。gitignoreファイルに`config.yaml`を追加し、APIキーがリポジトリにコミットされないように注意してください。
    * `config.yaml`の例 (ホスト名は指定の例を使用):
        ```yaml
        clearml:
          api_host: "https://api.clear.ml"
          web_host: "https://app.clear.ml"
          files_host: "https://files.clear.ml"
          api_key: "YOUR_API_KEY" # ここはユーザーが自身のキーを入力
          api_secret_key: "YOUR_SECRET_KEY" # ここはユーザーが自身のシークレットキーを入力
        ```
* **必要なライブラリのリストアップと`requirements.txt`の作成:** プロジェクトで使用するすべてのPythonライブラリとそのバージョンを`requirements.txt`にまとめてください。

**2. データ分析と前処理 (EDA & Preprocessing):**

* 提供されたデータファイル (`train.csv`, `test.csv`) を読み込んでください。
* 詳細な探索的データ分析（EDA）を実施し、データの特性（欠損値、外れ値、特徴量間の相関、分布、ターゲット変数との関係など）を把握してください。EDAの結果と考察はClearMLに記録してください（例: プロット画像、テキストログ）。
* EDAの結果に基づき、適切な前処理戦略を立案・実行してください。これには以下のような処理が含まれますが、これらに限定されません。
    * 欠損値処理
    * カテゴリ特徴量のエンコーディング (One-Hot Encoding, Label Encoding, Target Encodingなど)
    * 数値特徴量のスケーリング (StandardScaler, MinMaxScalerなど)
    * 必要であれば特徴量エンジニアリング (新しい特徴量の生成、既存特徴量の変換など)
* 前処理パイプラインを構築し、学習データとテストデータに一貫して適用できるようにしてください。このパイプラインは再利用可能で、ウェブアプリケーションでも使用できるように設計してください。

**3. モデル設計と学習:**

* 評価指標である **ROC AUC** を最大化することを目標に、複数の機械学習モデルを検討・実装してください (例: LightGBM, XGBoost, RandomForest, CatBoost、ロジスティック回帰など)。アンサンブル手法も検討してください。
* 適切な交差検証戦略（例: StratifiedKFold、ターゲットの分布を考慮）を実装して、モデルの汎化性能（ROC AUC）を評価してください。
* ハイパーパラメータチューニングを実施してください（例: Optuna、Grid Search、Random Searchなど）。チューニングの目的関数はROC AUCとしてください。チューニングの過程と結果もClearMLに記録してください。
* 最終的なモデル（またはアンサンブルモデル）を学習させ、ClearMLにモデルアーティファクトとして保存してください。

**4. 予測とSubmissionファイル生成:**

* 学習済みモデルを使用して、テストデータ(`test.csv`)に対する予測（陽性クラスの確率）を行ってください。
* テストデータへの前処理は、学習データに適用したものと全く同じ手順で行ってください。
* コンペティションの規定（`sample_submission.csv`の形式、カラム名 `id` と `target`、IDの順序、ターゲットの確率値など）を**厳密に確認し**、それに従ったSubmissionファイルを生成してください。ファイル名は `submission.csv` としてください。

**5. ウェブアプリケーション開発:**

* 学習済みモデル（ClearMLからダウンロード、またはローカルに保存されたもの）と前処理パイプラインを使用して、特徴量を入力すると予測結果（陽性クラスの確率と、可能であれば予測クラス）を出力するシンプルなウェブアプリケーションを開発してください。フレームワークはStreamlitまたはFlaskを推奨します。
* **特徴量入力のユーザビリティ:**
    * ウェブアプリケーションのUIでは、ユーザーが各特徴量にどのような値を入力すべきか（例: 数値の範囲、カテゴリの選択肢、単位など）を明確かつ分かりやすく提示してください。
    * 可能であれば、各特徴量の簡単な説明や入力例を表示してください。`train.csv`に見られるような典型的な値の範囲やカテゴリを参考にしてください。
    * コンペティションデータセットの `train.csv` や `test.csv` に含まれるすべての（ID以外の）特徴量を入力できるようにしてください。
* アプリケーションの起動方法と使用方法を明確にしてください。

**6. READMEファイル作成 (日本語版 `README_ja.md` と英語版 `README.md`):**

* 両方のREADMEファイルに、以下の情報を詳細に記述してください。
    * **プロジェクト概要:** コンペティションの簡単な説明と目的（予測ターゲット、評価指標）。
    * **コンペティション情報源:** (元URL: `https://www.kaggle.com/competitions/playground-series-s5e6/`)
    * **ディレクトリ構造:** プロジェクトの主要なファイルとフォルダの説明。
    * **セットアップ手順:**
        * 必要なライブラリのインストール方法 (`requirements.txt` を使用)。
        * ClearMLの設定方法（`config.yaml` の作成とAPIキーの設定について）。
    * **処理フロー:**
        * **データ:** 使用データの説明（train, test, sample_submission）。特徴量の概要（種類、数など）。ターゲット変数と評価指標（ROC AUC）。
        * **前処理 (Preprocessing):** 実行した具体的な前処理ステップとその理由。
        * **モデル設計 (Model Design):**
            * 採用したモデルアーキテクチャ（例: LightGBM）。
            * なぜそのモデルを選択したかの簡単な理由。
            * 主要なハイパーパラメータ（またはチューニング範囲と最終値）。
        * **学習 (Training):** 学習プロセス、交差検証の方法とスコア。
    * **実行方法:**
        * **学習パイプラインの実行手順:** モデルを再学習させるためのコマンドやスクリプト実行方法。
        * **Submissionファイルの生成手順:** `submission.csv` を生成するためのコマンドやスクリプト実行方法。
    * **ウェブアプリケーションの使い方:**
        * アプリケーションの起動コマンド。
        * 特徴量の入力方法と予測結果の確認方法。
    * **ClearMLの活用:** ClearMLで何をトラッキングしているかの簡単な説明。

**7. コードの品質と構成:**

* コードは適切にコメントを付け、モジュール化し、理解しやすいように記述してください。
* Pythonの標準的なコーディング規約（PEP 8など）に従うことを推奨します。
* エラーハンドリングを適切に行ってください。
* プロジェクトルートに `.gitignore` ファイルを作成し、不要なファイル（`__pycache__`, `config.yaml`, データファイル（もし大きい場合）、環境ファイルなど）がGitリポジトリに含まれないようにしてください。

**成果物:**

1.  すべてのソースコード（Pythonスクリプト、Jupyter Notebookなど）。
2.  `requirements.txt` ファイル。
3.  `config.yaml.template` (APIキーを記入するためのテンプレートファイル、実際のキーは含めない)。
4.  `submission.csv` ファイル (コンペティション規定準拠、ROC AUCのための確率値を含む)。
5.  ウェブアプリケーションのコード。
6.  `README.md` (英語)。
7.  `README_ja.md` (日本語)。
8.  ClearMLへの実験ログ。

**開発の進め方:**

上記のステップを順番に進めてください。各ステップで不明な点や判断に迷う場合は、適宜質問してください。特にコンペティションのルールやデータの解釈については、慎重に確認してください。

---
